README.md

```diff
@@ -18,4 +18,129 @@ Also you have to configure the `github` maven repository. You can either include
     <id>github</id>
     <url>https://andrsuh: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@maven.pkg.github.com/andrsuh/tiny-event-sourcing</url>
 </repository>
-```
\ No newline at end of file
+```
+# Example of how to use library
+## Theory
+https://www.eventstore.com/event-sourcing - a good source of information about event sourcing and CQRS.
+## Example
+First, when we implement events sourcing pattern we have to define aggregates. In this example we will be having user aggregate.
+```kotlin
+@AggregateType(aggregateEventsTableName = "aggregate-user")
+data class UserAggregate(
+    override val aggregateId: String
+) : Aggregate {
+    override var createdAt: Long = System.currentTimeMillis()
+    override var updatedAt: Long = System.currentTimeMillis()
+
+    var userName: String = ""
+    var userLogin: String = ""
+    var userPassword: String = ""
+    lateinit var defaultPaymentId: UUID
+    lateinit var defaultAddressId: UUID
+    var paymentMethods = mutableMapOf<UUID, PaymentMethod>()
+    var deliveryAddresses = mutableMapOf<UUID, DeliveryAddress>()
+}
+```
+@AggregateType annotations makes library understand that this is the aggregate recognisable by library.
+Each aggregate has its own domain events. They're written in UserAggregateDomainEvents.kt.
+This is the example of one of the events:
+```kotlin
+@DomainEvent(name = USER_CREATED_EVENT)
+class UserCreatedEvent(
+    val userLogin: String,
+    val userPassword: String,
+    val userName: String,
+    val userId: String,
+    createdAt: Long = System.currentTimeMillis(),
+) : Event<UserAggregate>(
+    name = USER_CREATED_EVENT,
+    createdAt = createdAt,
+    aggregateId = userId,
+) {
+    override fun applyTo(aggregate: UserAggregate) {
+        aggregate.userLogin = userLogin
+        aggregate.userPassword = userPassword
+        aggregate.userName = userName
+    }
+}
+```
+@DomainEvent annotation is also from the library. It accepts name of the event.
+Each event accepts some parameters from the constructor, returns event that belongs to the aggregate:
+```kotlin
+Event<UserAggregate>
+```
+and has override method applyTo that does manipulations with aggregate state.
```

Comment:
we should highlight that applyTo method contains logs that of applying the changes that the event describes (because the event describes the fact of the change). It transits the aggregate state from state A (before the change) to the state B (after the change). If we take all the events (N events) from aggregate event log and iteratively call applyTo method starting with an empty aggregate state, it will go over N state transitions and will be in a most actual state.

Reply:
I've added this text to the readme: we should highlight that applyTo method contains logs that of applying the changes that the event describes (because the event describes the fact of the change). It transits the aggregate state from state A (before the change) to the state B (after the change). If we take all the events (N events) from aggregate event log and iteratively call applyTo method starting with an empty aggregate state, it will go over N state transitions and will be in a most actual state.